shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/STS/PoolReap.hs

```diff
@@ -101,25 +112,51 @@ instance
     ]
 
 poolReapTransition ::
+  forall era.
   HasField "_poolDeposit" (Core.PParams era) Coin =>
   TransitionRule (POOLREAP era)
 poolReapTransition = do
   TRC (pp, PoolreapState us a ds ps, e) <- judgmentContext
 
-  let retired = eval (dom (_retiring ps ▷ setSingleton e))
+  let retired :: Set (KeyHash 'StakePool (Crypto era))
+      retired = eval (dom (_retiring ps ▷ setSingleton e))
+      pr :: Map.Map (KeyHash 'StakePool (Crypto era)) Coin
       pr = Map.fromSet (const (getField @"_poolDeposit" pp)) retired
+      rewardAcnts :: Map.Map (KeyHash 'StakePool (Crypto era)) (RewardAcnt (Crypto era))
       rewardAcnts = Map.map _poolRAcnt $ eval (retired ◁ _pParams ps)
+      rewardAcnts_ :: Map.Map (KeyHash 'StakePool (Crypto era)) (RewardAcnt (Crypto era), Coin)
+      rewardAcnts_ = Map.intersectionWith (,) rewardAcnts pr
+      rewardAcnts' :: Map.Map (RewardAcnt (Crypto era)) Coin
       rewardAcnts' =
         Map.fromListWith (<+>)
           . Map.elems
-          $ Map.intersectionWith (,) rewardAcnts pr
+          $ rewardAcnts_
+      refunds :: Map.Map (Credential 'Staking (Crypto era)) Coin
+      mRefunds :: Map.Map (Credential 'Staking (Crypto era)) Coin
       (refunds, mRefunds) =
         Map.partitionWithKey
           (\k _ -> eval (k ∈ dom (_rewards ds)))
           (Map.mapKeys getRwdCred rewardAcnts')
       refunded = fold $ Map.elems refunds
       unclaimed = fold $ Map.elems mRefunds
 
+  tellEvent $
+    let rewardAcntsWithPool =
+          Map.foldlWithKey'
+            ( \acc sp (ra, coin) ->
+                Map.insertWith (Map.unionWith (<>)) (getRwdCred ra) (Map.singleton sp coin) acc
+            )
+            Map.empty
+            rewardAcnts_
+        (refundPools', unclaimedPools') =
+          Map.partitionWithKey
+            (\k _ -> eval (k ∈ dom (_rewards ds)))
+            rewardAcntsWithPool
```

Comment:
pretty sure this is at least an order of magnitude slower than it needs to be, but it was the easiest way for my fried brain to reason about constructing this at the time. might be worth revisiting later : )